src/test/kotlin/lotto/domain/PurchaseSessionTest.kt

```diff
@@ -0,0 +1,106 @@
+package lotto.domain
+
+import lotto.exceptions.LottoException
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class PurchaseSessionTest {
+    @Test
+    fun `should return a deep copy`() {
+        val session1 = PurchaseSession(amount = 1)
+        val session2 = session1.copy(manualTicketsNumber = 10)
+        assertThat(session2).isNotEqualTo(session1)
+        assertThat(session2.amount).isEqualTo(session1.amount)
```

Comment:
```suggestion
        assertThat(session2).isNotEqualTo(session1)
            .extracting { it.amount }.isEqualTo(session1.amount)
```

This comment is optional.
I just wanted to show you there's a lot of cool tools that allow us to assert complex situations,
and we can chain them together.
By knowing them, we can tackle harder test scenarios in the future 👍

Reply:
That's amazing, thanks for the tip! With Kotlin and JUnit, there are so many options that it can sometimes feel a bit overwhelming😅